### ISSUE TITLE
Big list of small things

### ISSUE THREAD

**Issue by AntonyGarand** [wintergatan-community/virtual-mmx#42]
Hey!

I'd like to start contributing to the project, but while getting familiar with the codebase I'd prefer starting with smaller issues instead of features, and I couldn't find any.   
As such, before I (or anyone else) start working on these, I thought I'd make a list of small improvements I'd like the project to have, opening the discussion on these!  

# Issues

## Tooltips
Right now, I have no idea what most of the actions are doing, and even the ones I think I know aren't quite clear.   
I think having small tooltips on the actions, most likely using a dependency such as [react-tooltip](https://www.npmjs.com/package/react-tooltip), will improve the new user experience.

## Quick start guide / Documentation
Documentation for the overall usage of the machine.  
Perhaps also documentation for more advanced information specific parts.  

This would probably start as a wiki page, but can added to the website pretty easily.

## ~~Cursor: pointer~~ [Done]
Setting the cursor to a pointer wherever we can interact with objects
Subtask: Improve the "clickable" area of certain areas.

Ex: On the programming wheel, while hovering an area, the background gets darker, but barely.   
Then, once a pin is added, you can't click on the whole region anymore, instead you must click exactly on the pin, without visual indications.

![image](https://user-images.githubusercontent.com/9059487/99162256-b3196500-26c9-11eb-9446-cd8bc0aba24a.png)
![image](https://user-images.githubusercontent.com/9059487/99162272-e956e480-26c9-11eb-82a1-8d973c30c7d7.png)
![image](https://user-images.githubusercontent.com/9059487/99162274-eb20a800-26c9-11eb-9026-c093983a2275.png)

## Resizing and repositioning areas
While the app resizes itself depending on the original window size, and when increasing it, the programming wheel doesn't resize when making the window size smaller, making it impossible to access all sections.  
In addition, on wide screens, the position of the programming pins doesn't match the proper position, they remain in the middle and on the left, making it awkward looking.
I'd replace it to 33%/66%, while perhaps keeping them in their current position on smaller screens as needed.
![image](https://user-images.githubusercontent.com/9059487/99162352-ed373680-26ca-11eb-82ef-7c3a74ff8449.png)

## Headers for the programming wheel
The current wheel doesn't have headers, and while there is a color distinction between the various instruments, focusing on a specific note requires manually counting it. Having a header would make this easier

## ~~Can't write on the first note?~~ [Done]
This is more of a bug to fix, but I can't write on the first note, where the first e6 is currently. I can remove it, but not write to it

## End of the programming wheel: Identify looping section
At the bottom of the programming wheel, instead of "cutting", there are few sections with the notes of the start, most likely to facilitate writing the melody.  
It's currently not identified, other than not having the hover effect. I'd like to make the section delimiter a different color, or perhaps some other form of identification.

In addition, you can remove pins from there, but not add them. Some consistency would be nice, so I would remove the possibility of removing the pins.

## Progress bar
This one might be a feature instead of a smaller thing, but I'll still throw it in here.

At the moment, when playing the song, while there is a green line showing the current progress, there isn't an easy way to follow the progress of the song, both on the wheel, and overall.  
I would add a progress indicator, and while I'm currently not sure of the best way, I have few ideas:
1. Follow the green bar / Autoscroll
This would make it easier to find out where we are on the wheel itself
2. Make a scrollbar, easier navigation on the wheel
3. Make a progress bar somewhere, where we can see where we are on the wheel, as well as the current amount of rotations

# Summary
These are many small issues I saw when using the app the first couple of times, and I'm sure many others are experiencing it.  
Feel free to discuss the details and relevance of any of these, or giving me the green light so I can start working on them!

Also, what's up with the current features? Most of them are assigned, but little work has been done on them for few months now.  Are they stale, and therefore can be taken over?

**Comment by mozi-h** [comment]
I agree with all, but the Tooltip one. It adds bloat to the code, so the question is: Is it worth it?
The debugging tools currently available are Tooltips because of Doc-Comments (if created) and the React debugging browser extensions.

Ultimately, as I'm not actively developing (just meta-housekeeping) on the Project, it's your (plural) preference.

**Comment by AntonyGarand** [comment]
> It adds bloat to the code  

I'm not sure what you mean here, as overall it would only add few attributes to the jsx: ` <div data-tip="hello world">`
The feature itself would be nice, although we might want it to be toggleable, as advanced users might find them annoying if there are too many.

To be clear, this is the type of Tooltip I'm talking about: 
![image](https://user-images.githubusercontent.com/9059487/99193226-a9cdde00-2745-11eb-8833-f8f4d1632c26.png)

**Comment by FelixWohlfrom** [comment]
Maybe to make it even simpler, we could also use the default [html "title" attribute](https://www.w3schools.com/htmL/tryit.asp?filename=tryhtml_attributes_title), which might not look too fancy, but also does not add additional dependencies and still might offer some help for users that might not be familiar with the icons.

**Comment by AntonyGarand** [comment]
While the title attribute would work, as it requires keeping the mouses over for a certain period, and can't be customized nor disabled, I'd rather use a generic tooltip library.  
I agree the extra dependency will make things a bit more bloated, but it does give us more control on the result
